src/transformers/models/dab_detr/modeling_dab_detr.py

```diff
@@ -0,0 +1,2689 @@
+        if cross:
+            value_states = (
+                value_states.contiguous().view(-1, batch_size * self.num_heads, self.v_head_dim).transpose(0, 1)
+            )
+        else:
+            value_states = value_states.view(*v_proj_shape)
+
+        source_len = key_states.size(1)
+
+        attn_weights = torch.bmm(query_states, key_states.transpose(1, 2))
+
+        if attn_weights.size() != (batch_size * self.num_heads, target_len, source_len):
+            raise ValueError(
+                f"Attention weights should be of size {(batch_size * self.num_heads, target_len, source_len)}, but is"
+                f" {attn_weights.size()}"
+            )
+
+        if key_padding_mask is not None:
+            attn_weights = attn_weights.view(batch_size, self.num_heads, target_len, source_len)
+            attn_weights = attn_weights.masked_fill(
+                key_padding_mask.unsqueeze(1).unsqueeze(2),
+                float("-inf"),
+            )
+            attn_weights = attn_weights.view(batch_size * self.num_heads, target_len, source_len)
+        # TODO: attention.py line 381 -- Numerical stability
+        attn_weights = nn.functional.softmax(attn_weights - attn_weights.max(dim=-1, keepdim=True)[0], dim=-1)
+
+        if output_attentions:
+            # this operation is a bit awkward, but it's required to
+            # make sure that attn_weights keeps its gradient.
+            # In order to do so, attn_weights have to reshaped
+            # twice and have to be reused in the following
+            attn_weights_reshaped = attn_weights.view(batch_size, self.num_heads, target_len, source_len)
+            attn_weights = attn_weights_reshaped.view(batch_size * self.num_heads, target_len, source_len)
+        else:
+            attn_weights_reshaped = None
+
+        attn_probs = nn.functional.dropout(attn_weights, p=self.dropout, training=self.training)
+
+        attn_output = torch.bmm(attn_probs, value_states)
+
+        if attn_output.size() != (batch_size * self.num_heads, target_len, self.v_head_dim):
+            raise ValueError(
+                f"`attn_output` should be of size {(batch_size, self.num_heads, target_len, self.v_head_dim)}, but is"
+                f" {attn_output.size()}"
+            )
+
+        attn_output = attn_output.transpose(0, 1).contiguous().view(target_len, batch_size, self.out_dim)
+
+        attn_output = self.out_proj(attn_output)
+
+        return attn_output, attn_weights_reshaped
+
+
+# Modified from transformers.models.detr.modeling_detr.DetrEncoderLayer with DetrEncoderLayer->DABDETREncoderLayer,DetrConfig->DABDETRConfig
+class DABDETREncoderLayer(nn.Module):
+    def __init__(self, config: DABDETRConfig):
+        super().__init__()
+        self.embed_dim = config.d_model
+        self.self_attn = nn.MultiheadAttention(
+            self.embed_dim, config.encoder_attention_heads, dropout=config.attention_dropout
+        )
```

Comment:
There are two different implementations of MultiHeadAttention in ConditionalDetr HF and in the official implementations. One is the original implementation (DetrAttention is the name of the function in the ConditionalDetr HF code), which is equivalent to nn.MultiHeadAttention, and a modified version where the projections are not part of the init function—they come externally, allowing the model to jointly attend to information from different representation subspaces (DABDETRAttention and ConditionalDetrAttention in HF codes). I used the PyTorch nn.MultiHeadAttention module (in the encoder module) since I didn't see the point of having ~50 lines of code when I can simplify it with a single line. I also saw that in other files, like in the OneFormer or Bridgetower implementations, they used the original PyTorch module, so it just reinforced my idea. 
Should I use the other module?